src/components/reporter/widgets/SelectField.js

```diff
@@ -2,32 +2,54 @@ import React, { useState } from "react";
 import clsx from "clsx";
 import * as styles from "./Widgets.module.scss";
 
-const SelectField = ({ name, options, onChange, isInvalid }) => {
-  const [value, setValue] = useState("")
+const SelectField = ({ name, options, onChange, isInvalid, contest }) => {
+  const [value, setValue] = useState("");
+  const [qaMessage, setQaMessage] = useState(false);
 
   const handleChange = (e) => {
     setValue(e.target.value);
     onChange(e);
-  }
+    const { name, value } = e.target;
+
+    if (
```

Comment:
It's not ideal to have logic in our select component that is so specific to one particular implementation. The component shouldn't know anything about the data passed into it. It should be reusable, and therefore as generic as possible. It would be better to move this logic into the form that uses the component.

The same is true for the textarea field, which I added the markdown preview to. Now I want to use the textarea in a different place in my branch, but I don't want the markdown preview, so I'm going to have to refactor. If we generally try to make components that are reusable and generic, we can avoid that type of problem.

That being said, I know it is a little more complicated to move this right now, and this issue is a high priority, so I'm going to approve this as-is and create a new issue for refactoring that we can work on when we have the time.